apps/synapse-portal-framework/src/components/Search.tsx

```diff
@@ -0,0 +1,59 @@
+import React from 'react'
+import { TextField, Button, InputAdornment } from '@mui/material'
+
+type SearchProps = {
+  searchText: string
+  setSearchText: (value: string) => void
+  onSearch: (searchText: string) => void
+}
+
+const Search = ({ searchText, setSearchText, onSearch }: SearchProps) => {
```

Comment:
If you moved the searchText state variable to this new Search component, would you only need to pass in onSearch()?

Reply:
Whoops, yeah, defining that state in every component that uses the Search component would be a pain :sweat_smile: